scripts/export_costs_from_partial_run.py

```diff
@@ -0,0 +1,173 @@
+
+# File path for extracted cost matrix
+output_path = r"E:\otp4gb-py\trse_walk_test\costs\PM\test_metrics.csv"
+
+
+#### Functions
+class AggregationMethod(enum.StrEnum):
+    """Method to use when aggregating itineraries."""
+
+    MEAN = enum.auto()
+    MEDIAN = enum.auto()
+
+
+class CostResults(pydantic.BaseModel):
+    """Cost results from OTP saved to responses JSON file."""
+
+    origin: routing.Place
+    destination: routing.Place
+    plan: Optional[routing.Plan]
+    error: Optional[routing.RoutePlanError]
+    request_url: str
+
+
+def _matrix_costs(result: CostResults) -> dict:
+    matrix_values: dict[str, str | int | float | datetime.datetime] = {
+        "origin": result.origin.name,
+        "destination": result.destination.name,
+        "origin_id": result.origin.id,
+        "destination_id": result.destination.id,
+        "origin_zone_system": result.origin.zone_system,
+        "destination_zone_system": result.destination.zone_system,
+    }
+
+    if result.plan is None:
+        return matrix_values
+
+    matrix_values["number_itineraries"] = len(result.plan.itineraries)
+    if matrix_values["number_itineraries"] == 0:
+        return matrix_values
+
+    stats = [
+        "duration",
+        "walkTime",
+        "transitTime",
+        "waitingTime",
+        "walkDistance",
+        "otp_generalised_cost",
+        "transfers",
+        "generalised_cost",
+    ]
+    for s in stats:
+        values = []
+        for it in result.plan.itineraries:
+            val = getattr(it, s)
+            # Set value to NaN if it doesn't exist or isn't set
+            if val is None:
+                val = np.nan
+            values.append(val)
+
+        matrix_values[f"mean_{s}"] = np.nanmean(values)
+
+        if matrix_values["number_itineraries"] > 1:
+            matrix_values[f"median_{s}"] = np.nanmedian(values)
+            matrix_values[f"min_{s}"] = np.nanmin(values)
+            matrix_values[f"max_{s}"] = np.nanmax(values)
+            matrix_values[f"num_nans_{s}"] = np.sum(np.isnan(values))
+
+    matrix_values["min_startTime"] = min(i.startTime for i in result.plan.itineraries)
+    matrix_values["max_startTime"] = max(i.startTime for i in result.plan.itineraries)
+    matrix_values["min_endTime"] = min(i.endTime for i in result.plan.itineraries)
+    matrix_values["max_endTime"] = max(i.endTime for i in result.plan.itineraries)
+
+    return matrix_values
+
+
+def _write_matrix_files(
+        data: list[dict[str, float]],
+        matrix_file: pathlib.Path,
+) -> None:
+    matrix = pd.DataFrame(data)
+
+    if type(matrix_file) == str:
+        matrix_file = Path(matrix_file)
+
+    metrics_file = matrix_file.with_name(matrix_file.stem + "-metrics.csv")
+    matrix.to_csv(metrics_file, index=False)
+    print("Written cost metrics to {}\nThis can be used to create a lookup of trips requested".format(metrics_file))
```

Comment:
Import the `cost` module instead of copying the code here

Reply:
Addressed in [2bcb0cb](https://github.com/Transport-for-the-North/otp4gb-py/commit/2bcb0cbeab8c5102221a03f1a7c77665e5435e4e). See https://github.com/Transport-for-the-North/otp4gb-py/pull/4#discussion_r1255600822